### PR TITLE
Expand on the idea of working together

### PR DESCRIPTION
**Pull request by living-etc**

### CODE REVIEW

File: principles/working-together.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+It doesn't mean "mobbing until you die". Mobbing is a tool in our toolbox, and is only appropriate in some situations.
+
+It doesn't mean you must have all roles involved in all work. Why have a designer sat in a mob that's renewing a TLS certificate? That just leads to disengagement from the team by people who don't have anything to add.
+
+It doesn't mean collaborators must stick together throughout the work. The point of collaboration is knowledge transfer. Once knowledge has been transfered, the collaborators are free to go their separate ways.
+
+It doesn't mean six people sit around a screen while one of them spends an afternoon reading some documentation.
+
+## A note on "mobbing until you die"
+Mobbing is good at sharing information between the members of the team about how the solution to a problem is implemented, but it is a very poor tool for solving the problem. Mobbing to solve a problem is just brainstorming, and that's been [long discredited as an effective vehicle for creative problem solving][Beyond Productivity Loss in Brainstorming Groups].
```
Comment (pauldambra) [inline principles/working-together.md:23]: was this meant to be a link?

File: principles/working-together.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Working together over working alone
+
+## Why is this important to us
+The most important part of our job is problem solving. Writing code is least important part, since that's only implementing the solution to a problem. The important bit happens before that.
+
+There are many different domains that compose the solution to a problem: business processes, security, operability, UI/UX. The software we write to generate value for the Co-op is the sum of these concerns, therefore that cannot be considered in isolation from each other.
+
+## What does this mean?
```
Comment (pauldambra) [inline principles/working-together.md:8]: probably needs more about what working together means

not just mobbing but...

File: principles/working-together.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Working together over working alone
+
+## Why is this important to us
+The most important part of our job is problem solving. Writing code is least important part, since that's only implementing the solution to a problem. The important bit happens before that.
```
Comment (pauldambra) [inline principles/working-together.md:4]: I think there are more reasons to work together... e.g. keeping each other honest


maybe there are several "working together" pages needed to cover each?

File: principles/working-together.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+It doesn't mean six people sit around a screen while one of them spends an afternoon reading some documentation.
+
+## A note on "mobbing until you die"
+Mobbing is good at sharing information between the members of the team about how the solution to a problem is implemented, but it is a very poor tool for solving the problem. Mobbing to solve a problem is just brainstorming, and that's been [long discredited as an effective vehicle for creative problem solving][Beyond Productivity Loss in Brainstorming Groups].
+
+Mobbing is about writing code, and as described above, writing code is the least important thing we do.
+
+## If we're not mobbing from dawn until dusk, what are we supposed to do?
+The rest of the work that makes up the job of an engineer.
```
Comment (pauldambra) [inline principles/working-together.md:28]: do we need things that make up the job of a !engineer too? Or is this repo just about engineering?

File: principles/working-together.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+In our field, these include, and are not limited to:
+- Pairing
+- Chatting over a coffee
+- Code reviews
+- Pairing/mobbing briefly to agree a design, before separating to implement it
+- Documenting solutions for both others and your future self
+
+
+[Beyond Productivity Loss in Brainstorming Groups]: https://www.sciencedirect.com/science/article/pii/S006526011043004X
```
Comment (pauldambra) [inline principles/working-together.md:47]: Ah, I've never seen markdown links like that, you live and learn :)

File: principles/working-together.md (hunk 4 above)
Comment (living-etc) [reply principles/working-together.md:28]: I assumed it was just engineering, but that's a good point.

I'll update this.

File: principles/working-together.md (hunk 3 above)
Comment (living-etc) [reply principles/working-together.md:4]: I think those reasons should be listed here for now, and only add extra structure when this page becomes too cumbersome.

Could expand on those reasons once this is merged, or I can add you to my fork of the repo and you can update the branch.

Comment (andylongshaw) [reply principles/working-together.md:4]: I don't think you can say "the least important part". I wouldn't write that "design is the least important part" as I think that could come across as disrespectful. I think you can say that code is only one (possibly small) part of the solution.

File: principles/working-together.md (hunk 1 above)
Comment (andylongshaw) [reply principles/working-together.md:23]: I think you have a point about not mobbing everything but this reads quite negatively. There are good things about mobbing and there are also anti-patterns. Some teams successfully use mobbing as their default way of working so I don't think you can say "The point of collaboration is knowledge transfer. Once knowledge has been transfered, the collaborators are free to go their separate ways." or point off to a study about brainstorming. There is more to it than solving the immediate problem, such as knowledge transfer about the architecture, codebase, domain, etc. I'm happy to pair on this wording if that would be helpful?

Comment (living-etc) [reply principles/working-together.md:23]: I think that would be a good idea. I'll pop a thing in the calendar.

Comment (keithb-coop) [reply principles/working-together.md:23]: > Mobbing is good at sharing information between the members of the team about how the solution to a problem is implemented, but it is a very poor tool for solving the problem.

I can see how someone would come to this conclusion based on thing that happens here that we call "mobbing".

It's the opposite of what I understand to be the intention of "Mob"/"Ensemble" work, which is exactly to emphasise collaborative, cross-functional problem solving, and to de-emphasise the implementation. After all, the implementation goes through the one keyboard, we choose to greatly throttle the rate of making changes to code and code-like assets in order to release attention to be used on problem solving.

What I see is a big hammer, collaborative cross-functional problem solving, being used on small task nuts. This probably is a waste. But! I suggest that the signal here is less the Ensemble work isn't effective, and more that our cross-functional collaborative problem solving teams aren't being asked to solve problems collaboratively across functions. They are being asked to do tasks to implement solutions.

File: principles/working-together.md (hunk 3 above)
Comment (keithb-coop) [reply principles/working-together.md:4]: I agree with Chris, writing code is the least important part of what we do. It remains absolutely critical, completely essentially, and if done at all should be done with great care and skill, while also being the least important part. I also agree with Andy that many folks are likely to be offended by this statement. One problem across the industry is the tendency of folks working as Software Engineers to over-strongly identify with the code that manifests their solutions, and to over-strongly identify writing more code with adding value. This isn't their fault, its a syndrome built into the education and training of developers across the industry. We won't fix that. So while I agree with Chris' sentiment (I think!?) I also agree with Andy that this isn't the most helpful language with which to express it.

**Pull request merged**